go/types/path.go

```diff
@@ -405,3 +433,74 @@ func getAnnotation(str string) (ann, rem string) {
 type keyIndexable interface {
 	setIntoKey(v bool) keyIndexable
 }
+
+// PositionPath is like IndexPath, but indexes at a position rather than a key
+// for Sets and Maps.
+type PositionPath struct {
+	// Index is the value of the position. It may be negative to indicate an
+	// index from the back. For example, an Index of -1 means the last element.
+	Index int64
+	// IntoKey: see IndexPath.IntoKey.
+	IntoKey bool
+}
+
+func NewPositionPath(idx int64) PositionPath {
+	return PositionPath{idx, false}
+}
+
+func NewPositionIntoKeyPath(idx int64) PositionPath {
+	return PositionPath{idx, true}
+}
+
+func (pp PositionPath) Resolve(v Value) Value {
+	var absIndex uint64
+	if col, ok := v.(Collection); !ok {
+		return nil
+	} else if absIndex, ok = getAbsoluteIndex(col, pp.Index); !ok {
+		return nil
+	}
+
+	switch v := v.(type) {
+	case List:
+		if pp.IntoKey {
```

Comment:
For the record, your comment:

> I don't think supporting @key here is needed.

got lost in the rebase, sorry. My response was along the lines of

> I agree it's not that useful, and rather bizarre, but it is consistent with the key indexing.

I'll probably remove it.